internal/providers/terraform/azure/event_hubs_namespace.go

```diff
@@ -0,0 +1,139 @@
+package azure
+
+import (
+	"fmt"
+
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+	"github.com/tidwall/gjson"
+)
+
+func GetAzureRMEventHubsNamespaceRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "azurerm_eventhub_namespace",
+		RFunc: NewAzureRMEventHubs,
+	}
+}
+
+func NewAzureRMEventHubs(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	var events *decimal.Decimal
+	sku := "Basic"
+	location := d.Get("location").String()
+
+	if d.Get("sku").Type != gjson.Null {
+		sku = d.Get("sku").String()
+	}
+	costComponents := make([]*schema.CostComponent, 0)
+	if u != nil && u.Get("events").Type != gjson.Null {
+		events = decimalPtr(decimal.NewFromInt(u.Get("events").Int()))
+		costComponents = append(costComponents, eventHubsCostComponent("Ingress events", location, sku, events))
+	}
+	if u == nil {
+		costComponents = append(costComponents, eventHubsCostComponent("Ingress events", location, sku, events))
+	}
+	costComponents = append(costComponents, eventHubsThroughPutCostComponent("Throughput unit (1 MB/s ingress, 2 MB/s egress)", location, sku))
+
+	if sku == "Standard" {
+		costComponents = append(costComponents, eventHubsKafkaCostComponent("Kafka endpoint", location, sku))
+		costComponents = append(costComponents, eventHubsCaptureCostComponent("Capture", location, sku))
+	}
+
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: costComponents,
+	}
+}
+
+func eventHubsCostComponent(name, location, sku string, quantity *decimal.Decimal) *schema.CostComponent {
+	if quantity != nil {
+		quantity = decimalPtr(quantity.Div(decimal.NewFromInt(int64(1000000))))
+	}
+	return &schema.CostComponent{
+		Name:            fmt.Sprintf("%s (%s)", name, sku),
+		Unit:            "1M events",
+		UnitMultiplier:  1,
```

Comment:
Should this be set to 1000000, since that's mongo shows the price per 1M and the unit is "1M events", and MonthlyQuantity is whatever `monthly_ingress_events` is set to (and if it's not set it then shows costs-depends-on-usage).

Reply:
if I set unitMultipler as 1000000 the cost is getting multiplied with 1000000 and cost is too high

Reply:
Plz see my latest commit for it, it looks ok in that one.